src/rendezvous/channels/MSC3903ECDHv1RendezvousChannel.ts

```diff
@@ -0,0 +1,279 @@
+        if (this.aesKey instanceof Uint8Array) {
+            const crypto = getCrypto();
+
+            const iv = crypto.randomBytes(32);
+            const cipher = crypto.createCipheriv("aes-256-gcm", this.aesKey as Uint8Array, iv, { authTagLength: 16 });
+            const ciphertext = Buffer.concat([
+                cipher.update(data, "utf8"),
+                cipher.final(),
+                cipher.getAuthTag(),
+            ]);
+
+            return JSON.stringify({
+                iv: encodeBase64(iv),
+                ciphertext: encodeBase64(ciphertext),
+            });
+        }
+
+        if (!subtleCrypto) {
+            throw new Error('Neither Web Crypto nor Node.js crypto are available');
+        }
+
+        const iv = new Uint8Array(32);
+        window.crypto.getRandomValues(iv);
+
+        const encodedData = new TextEncoder().encode(data);
+
+        const ciphertext = await subtleCrypto.encrypt(
+            {
+                name: "AES-GCM",
+                iv,
+                tagLength: 128,
+            },
+            this.aesKey as CryptoKey,
+            encodedData,
+        );
+
+        return JSON.stringify({
+            iv: encodeBase64(iv),
+            ciphertext: encodeBase64(ciphertext),
+        });
+    }
+
+    public async send(data: any) {
+        if (!this.olmSAS) {
+            throw new Error('Channel closed');
+        }
+
+        const stringifiedData = JSON.stringify(data);
+
+        if (this.aesKey) {
+            await this.transport.send('application/json', await this.encrypt(stringifiedData));
+        } else {
+            await this.transport.send('application/json', stringifiedData);
+        }
+    }
+
+    private async decrypt({ iv, ciphertext }: { iv: string, ciphertext: string }): Promise<any> {
+        if (!ciphertext || !iv) {
+            throw new Error('Missing ciphertext and/or iv');
+        }
+
+        const ciphertextBytes = decodeBase64(ciphertext);
+
+        if (this.aesKey instanceof Uint8Array) {
+            const crypto = getCrypto();
+            // in contrast to Web Crypto API, Node's crypto needs the auth tag split off the cipher text
+            const ciphertextOnly = ciphertextBytes.slice(0, ciphertextBytes.length - 16);
+            const authTag = ciphertextBytes.slice(ciphertextBytes.length - 16);
+            const decipher = crypto.createDecipheriv(
+                "aes-256-gcm", this.aesKey as Uint8Array, decodeBase64(iv), { authTagLength: 16 },
+            );
+            decipher.setAuthTag(authTag);
+            return decipher.update(encodeBase64(ciphertextOnly), "base64", "utf-8") + decipher.final("utf-8");
+        }
+
+        if (!subtleCrypto) {
+            throw new Error('Neither Web Crypto nor Node.js crypto are available');
+        }
+
+        const plaintext = await subtleCrypto.decrypt(
+            {
+                name: "AES-GCM",
+                iv: decodeBase64(iv),
+                tagLength: 128,
```

Comment:
[MDN](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/decrypt) doesn't have `tagLength` documented.  Is it necessary?

Reply:
Confusingly it is in the [AesGcmParams](https://developer.mozilla.org/en-US/docs/Web/API/AesGcmParams) doc on MDN instead.

Reply:
> Confusingly it is in the [AesGcmParams](https://developer.mozilla.org/en-US/docs/Web/API/AesGcmParams) doc on MDN instead.

Huh, strange. :shrug: